### DISCUSSION TITLE
Renovate replaces npm packages with @xmldom/xmldom instead of updating its versions AND the PRs have wrong titles

### DISCUSSION THREAD

**Discussion by gamell** [renovatebot/renovate#23793] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

Placeholder value, please select the correct response from the dropdown

### Describe the problem

We've seen a couple of instances of PRs opened by the renovate bot where it totally replaces some of our dependencies with `@xmldom/xmldom`. This is of course very surprising as we don't even use this package in our project. The two times we've seen this happen, renovate always replaces our dependencies with this foreign package.

An insidious detail is that the PR description is **correct**, it's only when you go to the file changes that you realize everything is messed up.

Needless to say, this dependency replacement totally breaks our project.

### PR description with obviously wrong versions the packages are updated to

<img width="862" alt="Screenshot 2023-08-09 at 10 24 57 AM" src="https://github.com/renovatebot/renovate/assets/2460215/061bca7c-fea0-44d1-9bc0-f7d5e2bc6df5">

### That does NOT correspond to the actual changes

<img width="391" alt="Screenshot 2023-08-09 at 10 25 26 AM" src="https://github.com/renovatebot/renovate/assets/2460215/7191dea5-1fc9-4313-8be4-696787554018">

Not sure it's cause by the same issue, but the titles of the PRs generated in the same renovate run are also wrong (see screenshot below):

<img width="912" alt="Screenshot 2023-08-09 at 10 34 36 AM" src="https://github.com/renovatebot/renovate/assets/2460215/f6239459-081c-44a2-8bf5-976d6f030298">

It's important to note that **this does not happen consistently** so not sure how to replicate it.

Our config:

```
{
  "packageRules": [
    {
      "groupName": "Dashboard dependencies",
      "matchFileNames": ["dashboard/**"],
      "extends": [
        ":app",
        "config:recommended",
        "npm:unpublishSafe",
        ":automergeStableNonMajor",
        ":automergeRequireAllStatusChecks",
        ":automergeTypes",
        ":automergeTesters",
        "group:react",
        "group:react-routerMonorepo"
      ],
      "automergeType": "branch",
      "timezone": "America/Chicago",
      "schedule": [
        "after 10pm every weekday",
        "before 5am every weekday",
        "every weekend"
      ],
      "prConcurrentLimit": 5,
      "reviewers": ["moovfinancial:frontend"],
      "packageRules": [
        {
          "matchDepTypes": ["devDependencies"],
          "automerge": true
        },
        {
          "groupName": "@sentry updates",
          "matchPackagePrefixes": ["@sentry/"]
        },
        {
          "groupName": "victory-* updates",
          "matchPackagePrefixes": ["victory-"]
        }
      ]
    }
  ]
}

```

### Relevant debug logs

<details><summary>Logs</summary>

```
{
            "currentValue": "36.6.11",
            "currentVersion": "36.6.11",
            "datasource": "npm",
            "depName": "victory-area",
            "depType": "dependencies",
            "fixedVersion": "36.6.11",
            "homepage": "https://formidable.com/open-source/victory",
            "lockedVersion": "36.6.11",
            "packageName": "victory-area",
            "prettyDepType": "dependency",
            "registryUrl": "https://registry.npmjs.org",
            "sourceUrl": "https://github.com/FormidableLabs/victory",
            "versioning": "node",
            "warnings": [],
            "updates": [
              {
                "updateType": "replacement",
                "newName": "@xmldom/xmldom",
                "newValue": "0.7.5",
                "branchName": "renovate/react-monorepo"
              }
            ]
          },
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by viceice** [comment]
please provide your full config

**Reply by viceice** [reply]
https://github.com/renovatebot/renovate/blob/90582383b93721492d743fb29f0425838ddc305e/lib/config/presets/internal/replacements.ts#L733

this should only replace xmldom packages

**Reply by gamell** [reply]
Just updated the config to the whole config we have right now in master.

Re: your last comment, I am not sure I was clear in my initial description, but we don't even use the `xmldom` package in our repo, and we don't want it. The problem is renovate replacing packages we actually use with that one automatically.

**Reply by gamell** [reply]
Also if you check the logs I provided you will see right there that renovate is _replacing_ our package with `@xmldom/xmldom`:

```
            "warnings": [],
            "updates": [
              {
                "updateType": "replacement",
                "newName": "@xmldom/xmldom",
                "newValue": "0.7.5",
                "branchName": "renovate/react-monorepo"
              }
```

**Reply by viceice** [reply]
your config is wrong never nest package rules.

because you extend `config:recommend` inside it, all replacement are applied to matching packages.

**Reply by rarkins** [reply]
Can we add a test to catch and error for this? @RahulGautamSingh
